client.go

```diff
@@ -302,15 +302,25 @@ func (c *Client) CreateDataset(options *CreateDatasetOptions) (datasetState *Dat
 	tickChan := time.NewTicker(tickerWaitingForVolume).C
 
 	for {
-		if s, err := c.GetDatasetState(p.DatasetID); err == nil {
+		var strErrDel string
+		s, err := c.GetDatasetState(p.DatasetID)
+		if err == nil {
 			return s, nil
 		} else if err != errStateNotFound {
-			return nil, err
+			errDel := c.DeleteDataset(p.DatasetID)
```

Comment:
Ok. I'd be inclined to remove this since it seems unrelated to the Kubernetes crash bug, but if you think it's useful I'll go with it.